File: MyStagePass.Model/Models/Ticket.cs
using QRCoder;

namespace MyStagePass.Model.Models
{
	public class Ticket
	{
		public int TicketID { get; set; }
		public int Price { get; set; }
		public int EventID { get; set; }
		public virtual Event Event { get; set; } = null!;
		public int PurchaseID { get; set; }
		public virtual Purchase Purchase { get; set; } = null!;
		public Event.TicketType TicketType { get; set; }
		public bool IsDeleted { get; set; } = false;
		public string GetTicketTypeName()
		{
			return TicketType switch
			{
				Event.TicketType.Regular => "Regular",
				Event.TicketType.Vip => "VIP",
				Event.TicketType.Premium => "Premium",
				_ => "Unknown"
			};
		}
	}
}

File: UI/mystagepass_mobile/lib/models/Ticket/ticket.dart
import 'package:json_annotation/json_annotation.dart';
import '../Event/event.dart';

part 'ticket.g.dart';

@JsonSerializable()
class Ticket {
  int? ticketID;
  int? price;
  int? eventID;
  int? purchaseID;
  int? ticketType;
  String? qrCodeData;
  bool? isDeleted;
  Event? event;

  Ticket({
    this.ticketID,
    this.price,
    this.eventID,
    this.purchaseID,
    this.ticketType,
    this.qrCodeData,
    this.isDeleted,
    this.event,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) => _$TicketFromJson(json);
  Map<String, dynamic> toJson() => _$TicketToJson(this);

  String getTicketTypeDisplay() {
    switch (ticketType) {
      case 1:
        return 'Regular';
      case 2:
        return 'VIP';
      case 3:
        return 'Premium';
      default:
        return 'Unknown';
    }
  }
}
